e2e/ctl_v3_test.go

```diff
@@ -105,6 +110,10 @@ func withNoStrictReconfig() ctlOption {
 	return func(cx *ctlCtx) { cx.noStrictReconfig = true }
 }
 
+func withFlagByEnv() ctlOption {
+	return func(cx *ctlCtx) { cx.flagByEnv = true }
```

Comment:
@xiang90 Think it's needed for these

```go
for k, v := range fmap {
	if cx.flagByEnv {
		ek := flags.FlagToEnv("ETCDCTL", k)
		os.Setenv(ek, v)
		cx.envMap[ek] = struct{}{}
	} else {
		cmdArgs = append(cmdArgs, fmt.Sprintf("--%s=%s", k, v))
	}
}
```

Otherwise, we can't tell if it should use env vars or flags--which will overwrite env vars?